cmd/thanos/main.go

```diff
@@ -5,6 +5,8 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
+	"github.com/improbable-eng/thanos/pkg/tracing/provider"
```

Comment:
Not fan of this package name. Can we put this into `tracing`?  (:

Reply:
In `objstore` we have `client` package. Can we do it consistently for tracing as well?